init.c

```diff
@@ -99,44 +99,49 @@ struct MyVar
 {
   char *name;
   char *value;
-  struct MyVar *next;
+  STAILQ_ENTRY(MyVar) entries;
 };
 
-static struct MyVar *MyVars;
+STAILQ_HEAD(MyVarList, MyVar);
+static struct MyVarList MyVars = STAILQ_HEAD_INITIALIZER(MyVars);
 
 void myvar_set(const char *var, const char *val)
 {
-  struct MyVar **cur = NULL;
-
-  for (cur = &MyVars; *cur; cur = &((*cur)->next))
-    if (mutt_str_strcmp((*cur)->name, var) == 0)
-      break;
+  struct MyVar *myv = NULL;
 
-  if (!*cur)
-    *cur = mutt_mem_calloc(1, sizeof(struct MyVar));
+  mutt_debug(2, " * myvar_set(%s, '<hidden>')\n", NONULL((char *) var));
 
-  if (!(*cur)->name)
-    (*cur)->name = mutt_str_strdup(var);
+  STAILQ_FOREACH(myv, &MyVars, entries)
+  {
+    if (mutt_str_strcmp(myv->name, var) == 0)
+    {
+      mutt_str_replace(&myv->value, val);
+      return;
+    }
+  }
 
-  mutt_str_replace(&(*cur)->value, val);
+  myv = mutt_mem_calloc(1, sizeof(struct MyVar));
+  myv->name = mutt_str_strdup(var);
+  myv->value = mutt_str_strdup(val);
+  STAILQ_INSERT_TAIL(&MyVars, myv, entries);
 }
 
 static void myvar_del(const char *var)
 {
-  struct MyVar **cur = NULL;
+  struct MyVar *myv = NULL;
   struct MyVar *tmp = NULL;
+  mutt_debug(2, " * myvar_del(%s)\n", NONULL((char *) var));
 
-  for (cur = &MyVars; *cur; cur = &((*cur)->next))
-    if (mutt_str_strcmp((*cur)->name, var) == 0)
-      break;
-
-  if (*cur)
+  STAILQ_FOREACH_SAFE(myv, &MyVars, entries, tmp)
```

Comment:
You don’t need the _SAFE version of the macro because you’re not iterating after deletion.